test/e2e/modelregistry_controller_e2e_test.go

```diff
@@ -0,0 +1,343 @@
+			}, timeout, interval).Should(BeTrue())
+
+			Expect(actualISVC.Labels[constants.ModelRegistryRegisteredModelIdLabel]).To(Equal(""))
+			Expect(actualISVC.Labels[constants.ModelRegistryModelVersionIdLabel]).To(Equal(""))
+			Expect(actualISVC.Finalizers[0]).To(Equal("modelregistry.opendatahub.io/finalizer"))
+
+		})
+
+		It("the controller should create InferenceService without specific model version in model registry", func() {
+			_, err := utils.Run(exec.Command("kubectl", "apply", "-f", InferenceServiceWithoutModelVersionPath))
+			Expect(err).ToNot(HaveOccurred())
+
+			var is *openapi.InferenceService
+			// incremental id
+			isId := "4"
+			Eventually(func() error {
+				is, err = modelRegistryClient.GetInferenceServiceById(isId)
+				return err
+			}, timeout, interval).ShouldNot(HaveOccurred())
+
+			Expect(strings.HasPrefix(*is.Name, inferenceServiceName)).To(BeTrue())
+			Expect(is.ServingEnvironmentId).To(Equal("3"))
+			Expect(is.RegisteredModelId).To(Equal(*registeredModel.Id))
+			Expect(is.ModelVersionId).To(BeNil())
+
+			By("checking that the controller has correctly put the InferenceService id label in the ISVC")
+			actualISVC := &kservev1beta1.InferenceService{}
+			Eventually(func() bool {
+				namespacedNamed := types.NamespacedName{Name: inferenceServiceName, Namespace: WorkingNamespace}
+				err := cli.Get(ctx, namespacedNamed, actualISVC)
+				if err != nil {
+					return false
+				}
+				_, ok := actualISVC.Labels[constants.ModelRegistryInferenceServiceIdLabel]
+				return ok
+			}, timeout, interval).Should(BeTrue())
+
+			Expect(actualISVC.Labels[constants.ModelRegistryRegisteredModelIdLabel]).To(Equal(""))
+			Expect(actualISVC.Labels[constants.ModelRegistryModelVersionIdLabel]).To(Equal(""))
+			Expect(actualISVC.Finalizers[0]).To(Equal("modelregistry.opendatahub.io/finalizer"))
+		})
+	})
+
+	When("ISVC is deleted from the cluster", func() {
+		var inferenceService *openapi.InferenceService
+		BeforeEach(func() {
+			// fill mr with some models
+			fillModelRegistryContent(modelRegistryClient)
+			// Ensure IDs in model registry are created in a specific order
+			Expect(registeredModel.GetId()).To(Equal("1"))
+			Expect(modelVersion.GetId()).To(Equal("2"))
+			Expect(modelArtifact.GetId()).To(Equal("1"))
+
+			// simulate ServingEnvironment creation
+			envName := WorkingNamespace
+			_, err := modelRegistryClient.UpsertServingEnvironment(&openapi.ServingEnvironment{
+				Name: &envName,
+			})
+			Expect(err).ToNot(HaveOccurred())
+
+			inferenceService, err = modelRegistryClient.UpsertInferenceService(&openapi.InferenceService{
+				Name:                 &versionName,
+				DesiredState:         openapi.INFERENCESERVICESTATE_DEPLOYED.Ptr(),
+				RegisteredModelId:    *registeredModel.Id,
+				ServingEnvironmentId: "3",
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(inferenceService.GetId()).To(Equal("4"))
+
+			_, err = utils.Run(exec.Command("kubectl", "apply", "-f", ServingRuntimePath1))
```

Comment:
If possible, use an environment variable for `kubectl`. Later, we may run this in an environment where only `oc` may be available.

Reply:
Makes sense, much more flexible